api/google_analytics_provider.py

```diff
@@ -3,24 +3,45 @@
 import unicodedata
 import urllib
 import re
+from flask.ext.babel import lazy_gettext as _
 from core.util.http import HTTP
+from core.model import (
+    ConfigurationSetting,
+    ExternalIntegration,
+    Session,
+    get_one,
+)
 
 class GoogleAnalyticsProvider(object):
-    INTEGRATION_NAME = "Google Analytics"
+    NAME = _("Google Analytics")
+
+    TRACKING_ID = "tracking_id"
+    DEFAULT_URL = "http://www.google-analytics.com/collect"
+
+    SETTINGS = [
+        { "key": ExternalIntegration.URL, "label": _("URL"), "default": DEFAULT_URL },
+    ]
+
+    LIBRARY_SETTINGS = [
+        { "key": TRACKING_ID, "label": _("Tracking ID") },
+    ]
     
-    @classmethod
-    def from_config(cls, config):
-        tracking_id = config[Configuration.INTEGRATIONS][cls.INTEGRATION_NAME]['tracking_id']
-        return cls(tracking_id)
+    def __init__(self, integration):
+        url_setting = ConfigurationSetting.for_externalintegration(ExternalIntegration.URL, integration)
+        self.url = url_setting.value or self.DEFAULT_URL
+        self.integration_id = integration.id
 
-    def __init__(self, tracking_id):
-        self.tracking_id = tracking_id
+    def collect_event(self, library, license_pool, event_type, time, **kwargs):
+        _db = Session.object_session(library)
+        integration = get_one(_db, ExternalIntegration, id=self.integration_id)
+        tracking_id = ConfigurationSetting.for_library_and_externalintegration(
+            _db, self.TRACKING_ID, library, integration,
+        ).value
```

Comment:
I changed Analytics to keep a dictionary of providers by library id, as well as a list of sitewide providers. I didn't create a LibraryAnalytics class since there wasn't anything for it to do except call the provider.